### DISCUSSION TITLE
What is the replacement for mcxdeblast?

### DISCUSSION THREAD

**Discussion by 0xaf1f** [micans/mcl#25] [Q&A] [answered]
mcl in bioconda was suddenly updated to the latest version 22.282 from the previous version 14.137 that the bioinformatics community had been using and it broke some tools that relied on mcl's  `mcxdeblast` [1].

- The manual for mcxdeblast is still online: https://micans.org/mcl/man/mcxdeblast.html
- The Changelog makes no mention of its removal
- I only see in the README that:
>    Please be advised that the TribeMCL module has been disabled.  With mcl's
>   new label streaming facilities it is possible to cluster directly from BLAST
>   files. Refer to the mcl website for examples.

But I can't find anything about this on the mcl website. What is the successor to the mcxdeblast workflow in the current mcl version?

1. https://github.com/nf-core/modules/issues/3828

**Comment by micans** [comment, marked as answer]
My apologies for this, it was quite reckless of me to remove mcxdeblast - a cleaning out of old code was taken too far. I will re-instate it in the next release, which is still some way off. For now, the best I can suggest is to copy it e.g. from https://micans.org/mcl/src/mcl-14-137/src/alien/oxygen/src/mcxdeblast and put it somewhere that's in your `$PATH` variable. I prefer something like `$HOME/local/bin`. It has no dependencies, so just the script is needed. In Debian the maintainers fixed the issue themselves and they do ship `mcxdeblast` with version `22-282`, should that be an option for you.

**Reply by 0xaf1f** [reply]
Many thanks for clarifying this. I'm contributing a [similar change](https://github.com/bioconda/bioconda-recipes/pull/44975) to the bioconda package so that these get included automatically with conda installations, too.

**Reply by micans** [reply]
That's great - thanks for contributing that change, much appreciated!
